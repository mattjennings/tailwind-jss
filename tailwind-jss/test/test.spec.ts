import { transformSync } from '@babel/core'

const presets = ['module:@babel/preset-react']
const plugins = [require('../src')]

test('simple property', () => {
  const res = transformSync(
    `export function App() {
      return <div tw={{ bg: 'blue-500' }}  />;
    }`,
    {
      presets,
      plugins,
    }
  )
  expect(res?.code).toContain(`className: "bg-blue-500"`)
})

test('nested properties', () => {
  const res = transformSync(
    `export function App() {
      return <div tw={{ space: { x: 1, y: 1} }}  />;
    }`,
    {
      presets,
      plugins,
    }
  )
  expect(res?.code).toContain(`className: "space-x-1 space-y-1"`)
})

test('boolean properties', () => {
  const res = transformSync(
    `export function App() {
      return <div tw={{ rounded: true, border: { border: true, transparent: true }, something: false, bg: { 'indigo-500': true } }}  />;
    }`,
    {
      presets,
      plugins,
    }
  )
  expect(res?.code).toContain(
    `className: "rounded border border-transparent bg-indigo-500"`
  )
})

test('responsive', () => {
  const res = transformSync(
    `export function App() {
      return <div tw={{ bg: 'blue-500', 'hover:': { bg: 'blue-500' } }}  />;
    }`,
    {
      presets,
      plugins,
    }
  )
  expect(res?.code).toContain(`className: "bg-blue-500 hover:bg-blue-500"`)
})

// xtest('merges className', () => {
//   const res = transformSync(
//     `export function App() {
//       return <div tw={{ bg: 'blue-500' }} className="text-red-500" />;
//     }`,
//     {
//       presets,
//       plugins,
//     }
//   );
// });
