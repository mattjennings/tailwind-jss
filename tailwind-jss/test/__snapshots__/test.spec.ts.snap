// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translates nested properties to className 1`] = `
"export function App() {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    className: \\"space-undefined\\"
  });
}"
`;

exports[`translates to className 1`] = `
"export function App() {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    className: \\"bg-blue-500\\"
  });
}"
`;
